eng/build.ps1

```diff
@@ -116,6 +128,8 @@ if ($subset -eq 'help') {
   exit 0
 }
 
+Assert-InstalledDependency("Git")
```

Comment:
Git isn't a dependency. The repo should be buildable without any source control provider. https://github.com/dotnet/runtime/issues/47298 tracks protecting this scenario in CI.